### PR TITLE
Fixes #323: Enable use of service accounts

### LINKED ISSUE #323: Support for Kubernetes service accounts
As detailed at https://github.com/GoogleCloudPlatform/kubernetes/blob/master/docs/design/service_accounts.md.

Should be a simple case of reusing bearer token at `/var/run/secrets/kubernetes.io/servicaccount/token`.

### PR DESCRIPTION
**Pull request by jimmidyson**

### CODE REVIEW

**Comment by jimmidyson** [comment]
Fixes #320 too, or at least makes it easier to run on OpenShift.

**Comment by vishh** [comment]
@cadvisorJenkinsBot: ok to test

**Comment by vishh** [comment]
@jimmidyson: Can you add some documentation to go along with this PR? This will help support new users on OpenShift.

File: sources/kube_factory.go [hunk 1]
```diff
@@ -31,11 +32,13 @@ import (
...
-	defaultKubeletPort    = 10255
-	defaultKubeConfigFile = "/etc/kubernetes/kubeconfig/kubeconfig"
-	defaultKubeletHttps   = false
+	defaultApiVersion         = "v1beta3"
+	defaultInsecure           = false
+	defaultKubeletPort        = 10255
+	defaultKubeConfigFile     = "/etc/kubernetes/kubeconfig/kubeconfig"
+	defaultKubeletHttps       = false
+	defaultUseServiceAccount  = false
+	defaultServiceAccountFile = "/var/run/secrets/kubernetes.io/serviceaccount/token"
```
Comment (vishh) [inline sources/kube_factory.go]: Will `/var/run/secrets' be auto mounted for the heapster pod?

Comment (jimmidyson) [inline sources/kube_factory.go]: Yeah that's how service accounts work. You create a service account, a token is generated, then you can reference that service account in your pod spec & token is automounted.

Comment (vishh) [inline sources/kube_factory.go]: How about adding an example config for running heapster in OpenShift? Does it already exist elsewhere?

Comment (jimmidyson) [inline sources/kube_factory.go]: I'm working on it for OpenShift Origin docs & will either link to it from here or add it here too. There are some problems to work through as OpenShift is locked down right now so accessing /stats endpoint on the kubelet isn't possible... working on that too,

File: docs/source-configuration.md [hunk 2]
```diff
@@ -28,6 +28,20 @@ To run without auth file, use the following config:
...
+```
+cat <EOF | kubectl create -f -
+apiVersion: v1beta3
+kind: ServiceAccount
+metadata:
+  name: heapster
+EOF
+```
+
+You will then need to reference the service account in your Heapster pod spec.
```
Comment (vishh) [inline docs/source-configuration.md]: Can you provide a pointer to an example PodSpec or may be some pod that uses service accounts?

**Comment by vishh** [comment]
LGTM. Thanks @jimmidyson

**Pull request merged**